tls/extensions/s2n_client_supported_versions.c

```diff
@@ -71,6 +71,13 @@ int s2n_extensions_client_supported_versions_process(struct s2n_connection *conn
         uint8_t client_version_parts[S2N_TLS_PROTOCOL_VERSION_LEN];
         GUARD(s2n_stuffer_read_bytes(extension, client_version_parts, S2N_TLS_PROTOCOL_VERSION_LEN));
 
+        /* If the client version is outside of our supported versions, then ignore the value.
+         * S2N does not support SSLv2 except for upgrading connections. Since this extension is
+         * a TLS1.3 extension, we will skip any SSLv2 values. */
+        if (client_version_parts[0] <= 2 || client_version_parts[1] > 4) {
```

Comment:
But it won’t skip 0x04, 0x04 which is also invalid, right?

Just make the first part `!= 3`?

Reply:
You beat me to it. My next push has this check, and I updated the test as well.